inst/include/armadillo_bits/eop_aux.hpp

```diff
@@ -22,6 +22,8 @@
 //! use of the SFINAE approach to work around compiler limitations
 //! http://en.wikipedia.org/wiki/SFINAE
 
+#undef log1p
+
```

Comment:
We almost surely do not want to unconditionally undefine something in the upstream sources.

Reply:
This one is a workaround. I will make another PR if I succeed in changing the configure file instead of modifying the upstream armadillo code.

Here is my environment information:

```
> sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: aarch64-apple-darwin20.0.0 (64-bit)
Running under: macOS  11.2

Matrix products: default
BLAS:   /Users/xxx/miniforge3/envs/py38/lib/R/lib/libRblas.dylib
LAPACK: /Users/xxx/miniforge3/envs/py38/lib/R/lib/libRlapack.dylib

locale:
[1] C/UTF-8/C/C/C/C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.3 tools_4.0.3  
```
with (on Mac M1):

```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: arm64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
and

```
> packageVersion("Rcpp")
[1] ‘1.0.6’
```

I use the the latest CRAN/Github master version to install `RcppArmadillo` , then it reports an error

```
arm64-apple-darwin20.0.0-clang++ -std=gnu++11 -I"/Users/yifanyang/miniforge3/envs/py38/lib/R/include" -DNDEBUG  -I'/Users/yifanyang/miniforge3/envs/py38/lib/R/library/Rcpp/include' -D_FORTIFY_SOURCE=2 -isystem /Users/yifanyang/miniforge3/envs/py38/include -mmacosx-version-min=11.0 -I/Users/yifanyang/miniforge3/envs/py38/include  -I../inst/include  -fPIC  -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden  -fmessage-length=0 -isystem /Users/yifanyang/miniforge3/envs/py38/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/r-base-split_1615277992527/work=/usr/local/src/conda/r-base-4.0.3 -fdebug-prefix-map=/Users/yifanyang/miniforge3/envs/py38=/usr/local/src/conda-prefix  -c RcppArmadillo.cpp -o RcppArmadillo.o
In file included from RcppArmadillo.cpp:21:
In file included from ../inst/include/RcppArmadillo.h:31:
In file included from ../inst/include/RcppArmadilloForward.h:49:
In file included from ../inst/include/armadillo:374:
../inst/include/armadillo_bits/eop_aux.hpp:97:124: error: no member named 'Rlog1p' in namespace 'std'; did you mean 'log1p'?
  template<typename eT> arma_inline static typename   arma_integral_only<eT>::result log1p (const eT  x) { return eT( std::log1p(double(x)) ); }
                                                                                                                      ~~~~~^~~~~
                                                                                                                           log1p
/Users/yifanyang/miniforge3/envs/py38/lib/R/include/Rmath.h:75:15: note: expanded from macro 'log1p'
#define log1p Rlog1p
              ^
/Users/yifanyang/miniforge3/envs/py38/bin/../include/c++/v1/cmath:435:9: note: 'log1p' declared here
using ::log1p;
        ^
In file included from RcppArmadillo.cpp:21:
In file included from ../inst/include/RcppArmadillo.h:31:
In file included from ../inst/include/RcppArmadilloForward.h:49:
In file included from ../inst/include/armadillo:374:
../inst/include/armadillo_bits/eop_aux.hpp:98:115: error: no member named 'Rlog1p' in namespace 'std'; did you mean simply 'Rlog1p'?
  template<typename eT> arma_inline static typename       arma_real_only<eT>::result log1p (const eT  x) { return std::log1p(x);               }
                                                                                                                  ^~~~~~~~~~
                                                                                                                  Rlog1p
../inst/include/armadillo_bits/eop_aux.hpp:99:86: note: 'Rlog1p' declared here
  template<typename eT> arma_inline static typename         arma_cx_only<eT>::result log1p (const eT& x) { arma_ignore(x); return eT(0);       }
                                                                                     ^
/Users/yifanyang/miniforge3/envs/py38/lib/R/include/Rmath.h:75:15: note: expanded from macro 'log1p'
#define log1p Rlog1p
              ^
In file included from RcppArmadillo.cpp:21:
In file included from ../inst/include/RcppArmadillo.h:31:
In file included from ../inst/include/RcppArmadilloForward.h:49:
In file included from ../inst/include/armadillo:433:
../inst/include/armadillo_bits/fn_misc.hpp:170:21: error: no member named 'Rlog1p' in namespace 'std'; did you mean simply 'Rlog1p'?
    return (log_a + std::log1p(std::exp(negdelta)));
                    ^~~~~~~~~~
                    Rlog1p
/Users/yifanyang/miniforge3/envs/py38/lib/R/include/Rmath.h:74:9: note: 'Rlog1p' declared here
double  Rlog1p(double);
        ^
3 errors generated.
make: *** [/Users/yifanyang/miniforge3/envs/py38/lib/R/etc/Makeconf:181: RcppArmadillo.o] Error 1
ERROR: compilation failed for package ‘RcppArmadillo’
```